typed-racket-lib/typed-racket/optimizer/float.rkt

```diff
@@ -136,10 +142,10 @@
                           ;;   (Note: could allow for more args, if not next to each other, but
                           ;;    probably not worth the trouble (most ops have 2 args anyway))
                           (and (subtypeof? this-syntax -Flonum)
-                               (for/and ([a (in-syntax #'(fs ...))])
+                               (for/and ([a (in-syntax #'(fs ...))]
+                                         #:unless (subtypeof? a -Flonum))
```

Comment:
I don't think this is clearer.

Reply:
Yeah this one seems questionable. Filed jackfirth/resyntax#357.

Reply:
Undid this one and changed Resntax to only apply this rule if the last expression is a `(let ...)`.